addon/components/copy-button.js

```diff
@@ -1,39 +1,50 @@
 import Component from '@ember/component';
-import { set } from '@ember/object';
+import { set, action } from '@ember/object';
+import ClipboardJS from 'clipboard';
+
 import layout from '../templates/components/copy-button';
 
 const CLIPBOARD_EVENTS = ['success', 'error'];
 
-export default Component.extend({
-  layout: layout,
-  tagName: 'button',
-  classNames: ['copy-btn'],
-  attributeBindings: [
-    'clipboardText:data-clipboard-text',
-    'clipboardTarget:data-clipboard-target',
-    'clipboardAction:data-clipboard-action',
-    'buttonType:type',
-    'disabled',
-    'aria-label',
-    'title'
-  ],
-
-  /**
-   * @property {String} buttonType - type attribute for button element
-   */
-  buttonType: 'button',
+export default class extends Component {
+  layout = layout;
+  tagName = '';
 
   /**
    * @property {Boolean} disabled - disabled state for button element
    */
-  disabled: false,
+  disabled = false;
 
   /**
    * If true - scope event listener to this element
    * If false - scope event listener to document.body (ClipboardJS)
    * @property {Boolean} delegateClickEvent
    */
-  delegateClickEvent: true,
+  delegateClickEvent = true;
```

Comment:
can we just pass the element instance to clipboardJS? would that be easier?

Reply:
The behavior of ClipboardJS is different when you pass a selector vs an element.

With a selector you get event delegation where the click event is handled at `document.body`. With an element reference, the click handler is attached to the element.

See https://github.com/jkusa/ember-cli-clipboard/pull/35 for reference

Reply:
legit! is that situation in #35 described in documentation somewhere?